x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/elastic/response/ElasticInferenceServiceAuthorizationResponseEntity.java

```diff
@@ -134,7 +313,7 @@ public ElasticInferenceServiceAuthorizationResponseEntity() {
     }
 
     public ElasticInferenceServiceAuthorizationResponseEntity(StreamInput in) throws IOException {
-        this(in.readCollectionAsList(AuthorizedModel::new));
+        this(in.readCollectionAsList(AuthorizedEndpoint::new));
     }
 
     public static ElasticInferenceServiceAuthorizationResponseEntity fromResponse(Request request, HttpResult response) throws IOException {
```

Comment:
Not related to the changes in this PR, but the `request` parameter isn't used in this method, so it can be removed. There's also an unused no-arg constructor a few lines above this that can be removed.

Reply:
Yeah that's an unfortunate restriction of the framework we have.

```
@FunctionalInterface
public interface ResponseParser {
    InferenceServiceResults apply(Request request, HttpResult result) throws IOException;
}
```

That requires the request to be accepted in all response handlers. If we ever refactor we can decouple the authorization parsing from being a true inference response entirely and handle it then.